src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,538 @@
+		return metaData;
+	}
+
+	/**
+	* 
+	* @param {*} session 
+	*/
+	async logout(session) {
+		await super.logout();
+	}
+
+	/**
+	* 
+	* @param {*} metadata 
+	*/
+	async storeMetadata(metaData, params) {
+		const webUntisMetadataService = this.app.service('webuntisMetadata');
+
+		await Promise.all(metaData.map(async (entry) => {
+			const metadataResults = await webUntisMetadataService.find({ query: {
+				datasourceId: params.datasourceId,
+				teacher: entry.teacher,
+				class: entry.class,
+				subject: entry.subject
+			}, paginate: false });
+			const result = metadataResults[0];
+
+			if (result !== undefined) { // patch/replace existing
+				await webUntisMetadataService.patch(result._id, { state: entry.state, times: entry.times });
+			} else { // create new
+				await webUntisMetadataService.create(Object.assign(
+					{ datasourceId: params.datasourceId },
+					entry
+				));
+			}
+		}));
+	}
+
+	/**
+	* 
+	* @param {*} data 
+	*/
+	async createCourses(data, params) {
+		// Convert metadata to actual schulcloud courses
+		// And reflect changes in metadata store
+		const webUntisMetadataService = this.app.service('webuntisMetadata');
+
+		const importCondition = this.getImportConditionEvaluator(params);
+		const promises = data.map((entry) => {
+			if (importCondition(entry.state.toString())) {
+				return this.obtainAndUpdateCourseAndClass(entry)
+					.then(() => webUntisMetadataService.patch(entry._id, { state: 'imported' }))
+					.catch(() => webUntisMetadataService.patch(entry._id, { state: 'errored' }));
+			} else {
+				return webUntisMetadataService.patch(entry._id, { state: 'discarded' });
+			}
+		});
+
+		await Promise.all(promises);
+	}
+
+	/**
+	* Entry: {
+	*    datasourceId: this.data.datasourceId,
+	*    teacher: 'Renz',
+	*    class: '2a',
+	*    times: [ { weekDay, startTime, endTime, room }],
+	*    subject: 'mathe',
+	*    state: 'new',
+	* }
+	*/
+	async obtainAndUpdateCourseAndClass(entry) {
+		const courseService = this.app.service('courses');
+		const classService = this.app.service('classes');
+
+		const school = await this.getSchool(await this.getUser());
+
+		/**
+        * Mapping:
+        *
+        * Schul-Cloud: class, course (per class), lesson
+        * WebUntis: class, subject
+        * German: Klasse, Kurs, Fach, Schulstunde
+        */
+
+		// Get class
+		const scClasses = await classService.find({ query: { name: entry.class }, paginate: false });
+		
+		let scClass = scClasses[0];
+		if (scClass === undefined) {
+			// Create Schul-Cloud class?
+			const newClass = {
+				name: className,
+				schoolId: school._id,
+				nameFormat: 'static',
+				year: school.currentYear,
+			};
+			scClass = await classService.create(newClass);
+		}
+
+		// Get course
+		const scCourses = await courseService.find({
+			query: {
+				name: entry.subject,
+				classIds: scClass._id,
+				schoolId: school._id,
+			},
+			paginate: false,
+		});
+
+		scCourse = scCourses[0];
+		if (scCourse === undefined) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['scCourse' is not defined.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4061937)